crates/anvil-polkadot/tests/it/filters.rs

```diff
@@ -234,3 +243,743 @@ async fn test_filter_is_evicted() {
+    let decoded_data = SimpleStorage::ValueChanged::abi_decode_data(&data.0).unwrap();
+    // Assert the old value
+    assert_eq!(decoded_data.0, old);
+    // Assert the new value
+    assert_eq!(decoded_data.1, new);
+}
+
+#[tokio::test(flavor = "multi_thread")]
+async fn test_logs_filter_receives_new_logs() {
+    let anvil_node_config = AnvilNodeConfig::test_config();
+    let substrate_node_config = SubstrateNodeConfig::new(&anvil_node_config);
+    let mut node = TestNode::new(anvil_node_config.clone(), substrate_node_config).await.unwrap();
+
+    let alith = Account::from(subxt_signer::eth::dev::alith());
+    let alith_address = alith.address();
+
+    // Deploy contract
+    let contract_code = get_contract_code("SimpleStorage");
+    let tx_hash = node.deploy_contract(&contract_code.init, alith.address()).await;
+    unwrap_response::<()>(node.eth_rpc(EthRequest::Mine(None, None)).await.unwrap()).unwrap();
+    let receipt = node.get_transaction_receipt(tx_hash).await;
+    let contract_address = receipt.contract_address.unwrap();
+
+    // Create a new logs filter for the contract_address
+    let filter_id = unwrap_response::<String>(
+        node.eth_rpc(EthRequest::EthNewFilter(
+            Filter::new().address(Address::from(ReviveAddress::new(contract_address))),
+        ))
+        .await
+        .unwrap(),
+    )
+    .unwrap();
+    // Check that before interacting with the contract there are no logs
+    let logs = unwrap_response::<Vec<Log>>(
+        node.eth_rpc(EthRequest::EthGetFilterChanges(filter_id.clone())).await.unwrap(),
+    )
+    .unwrap();
+    assert!(logs.is_empty());
+
+    // Emit an event by calling setValue
+    let set_value_data = SimpleStorage::setValueCall::new((U256::from(511),)).abi_encode();
+    let call_tx = TransactionRequest::default()
+        .from(Address::from(ReviveAddress::new(alith_address)))
+        .to(Address::from(ReviveAddress::new(contract_address)))
+        .input(TransactionInput::both(set_value_data.into()))
+        .nonce(1);
+    node.send_transaction(call_tx).await.unwrap();
+    unwrap_response::<()>(node.eth_rpc(EthRequest::Mine(None, None)).await.unwrap()).unwrap();
+
+    let logs = unwrap_response::<Vec<Log>>(
+        node.eth_rpc(EthRequest::EthGetFilterChanges(filter_id.clone())).await.unwrap(),
+    )
+    .unwrap();
+
+    // Check the number of logs
+    assert_eq!(logs.len(), 1);
+    assert_eq!(logs[0].address, contract_address);
+    // Check that our topic is the ValueChanged event.
+    let event_hash = keccak_256(b"ValueChanged(address,uint256,uint256)");
+    assert_eq!(logs[0].topics[0], H256::from(event_hash));
+
+    assert_decoded_simple_storage_data(
+        logs[0].data.as_ref().unwrap(),
+        U256::from(0),
+        U256::from(511),
+    );
+
+    // Assert the changer address
+    let changer_topic = logs[0].topics[1].as_bytes();
+    let mut changer = [0u8; 20];
+    changer.copy_from_slice(&changer_topic[12..32]);
```

Comment:
isn't there a nicer way to use the abi types generated by the sol! macro for this? (as we do the abi encode/decode)